src/main/java/org/tron/program/Benchmark.java

```diff
@@ -0,0 +1,318 @@
+    /* ====================================================================== */
+    String params = "0000000000000000000000000000000000000000000000000000000000004e20" +
+        "00000000000000000000000000000000000000000000000000000000000007d0" +
+        "00000000000000000000000000000000000000000000000000000000000000c8";
+    byte[] triggerData = VMTimeBenchmarkUtils
+        .parseABI("timeBenchmark(uint256,uint256,uint256)", params);
+    TVMResult result = VMTimeBenchmarkUtils
+        .triggerContractAndReturnTVMResult(Hex.decode(OWNER_ADDRESS),
+            contractAddress, triggerData, 0, feeLimit, dbManager, null);
+
+    return result.getDuration();
+  }
+
+  public static TimeBenchmarkResult getTimeRatio()
+      throws ContractExeException, ReceiptCheckErrException, VMIllegalException, ContractValidateException, InterruptedException {
+
+    TimeBenchmarkResult result = timeBenchmark();
+
+    return result;
+  }
+
+  public static long getMem() {
+
+    com.sun.management.OperatingSystemMXBean os = (com.sun.management.OperatingSystemMXBean)
+        java.lang.management.ManagementFactory.getOperatingSystemMXBean();
+    long physicalMemorySize = os.getTotalPhysicalMemorySize();
+
+    return (long)(physicalMemorySize * 1.0 / 1024 / 1024 * 0.8);
+  }
+
+  public static boolean checkJavaVersion() {
+    String name = System.getProperty("java.runtime.name").toLowerCase();
+    if (name.indexOf("openjdk") != -1) {
+      return false;
+    }
+    String[] version = System.getProperty("java.vm.specification.version").split("\\.");
+    if (version.length < 2) {
+      return false;
+    } else if (Integer.valueOf(version[0]) * 10000 + Integer.valueOf(version[1]) < 10008) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  public static String readFile(String fileName) {
+
+    File file = new File(fileName);
+    if (null != file && file.canRead()) {
+      try {
+        Reader in = new FileReader(file);
+        StringWriter out = new StringWriter();
+        copy(in, out);
+        return out.toString();
+      } catch(IOException e) {
+        logger.info(e.getMessage());
+      }
+    }
+    return null;
+  }
+
+  public static void writeFile(String content, String fileName) {
+
+    try {
+      File file = new File(fileName);
+      Writer writer = new FileWriter(file);
+      writer.write(content);
+      writer.close();
+    } catch (IOException e) {
+      logger.info(e.getMessage());
+    }
+  }
+
+  public static void main(String[] args) {
+
+    boolean checkVersion = checkJavaVersion();
+
+    if (checkVersion) {
+      System.out.println("1. JAVA VERSION:\nsatisfied");
+    } else {
+      System.out.println("1. JAVA VERSION:\njava MUST be oracle jdk, and version >= 1.8");
+    }
+
+    try {
+      long mem = getMem();
+      String content = readFile("start.sh");
+      String newContent = content.replaceAll("java -jar", "java -Xmx" + mem + "m -jar");
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "content" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1546&issues=AWXsmUACxJR_uy9B78-O&open=AWXsmUACxJR_uy9B78-O)

[ISSUE_KEY]: AWXsmUACxJR_uy9B78-O